llvm/lib/Target/AMDGPU/AMDGPUTargetTransformInfo.cpp

```diff
@@ -344,9 +344,10 @@ unsigned GCNTTIImpl::getMinVectorRegisterBitWidth() const {
 unsigned GCNTTIImpl::getMaximumVF(unsigned ElemWidth, unsigned Opcode) const {
   if (Opcode == Instruction::Load || Opcode == Instruction::Store)
     return 32 * 4 / ElemWidth;
-  return (ElemWidth == 16 && ST->has16BitInsts()) ? 2
-       : (ElemWidth == 32 && ST->hasPackedFP32Ops()) ? 2
-       : 1;
+  return ElemWidth == 8                                ? 4
```

Comment:
Do all the TTI->get*Cost queries handled i8/vector i8 types?

Reply:
If we allow SLP to consider i8s for vectorization, then we are relying on the cost model to make the right decision of vectorizing certain sequcnes vs not. Given that we are relying on the cost model now to make accurate decisions, we should double check that the other cost queries are behaving as expected for i8s / vector i8s.

Since these cost queries are used elsewhere, I assume they already handle these types well, but it's worth double checking.

Reply:
This is only called in 2 places so I added an extra test to cover that.

Reply:
The TTI::get*Cost queries are called in many places. Can you please double check that the ones we have implemented in AMDGPUTargetTransformInfo either revert to the base handling or are explicitly customized for i8

Reply:
Why the cost queries? This function is only called in a handful of places and is not called by all cost queries?

Reply:
By changing the `getMaximimumVF` and `getNumberOfParts` we are effectively enabling SLPVectorization for i8s. Thus, the control for i8 vectorization has shifted from this function (and `getNumberOfParts`) to the cost queries. If the cost queries say the vector cost is lower, then we will vectorize. So, since we are transferring control, we should be sure that it was working as expected.

Reply:
I checked and no other arch specific changes (other than the ones in this patch) are affected by the changes to these functions for i8s.